fms/utils/serialization.py

```diff
@@ -124,10 +146,20 @@ def load_state_dict(
     if not model_path.is_dir():
         checkpoints = [model_path]
     else:
-        # TODO: add support for safetensors.
         checkpoints = sorted(model_path.glob("*.pth"))
         if not len(checkpoints):
             checkpoints = sorted(model_path.glob("*.bin"))
+        if not len(checkpoints):
+            checkpoints = sorted(model_path.glob("*.safetensors"))
```

Comment:
if someone specifies checkpoint_format='st' but the directory contains .bin files, won't this code still assign the bin file paths to `checkpoints`?
I think if we have a parameter we'll need to honor it everywhere. e.g. if someone specifies checkpoint_format='pt' then it should fail-fast when/if *.pth is empty.

Reply:
oh I totally forgot to re-implement this check, I was planning to enforce if given a concrete choice